service-base/src/main/java/org/eclipse/hono/service/management/Util.java

```diff
@@ -84,4 +107,55 @@ public static final Span newChildSpan(final String operationName, final SpanCont
         return newChildSpan(operationName, spanContext, tracer, tenantId, null, className);
     }
 
+    /**
+     * Creates a response based on generic result.
+     * <p>
+     * The behavior is as follows:
+     * <ol>
+     * <li>Set the status code on the response.</li>
+     * <li>If the status code represents an error condition (i.e. the code is &gt;= 400),
+     * then the JSON object passed in to the returned handler is written to the response body.</li>
+     * <li>Otherwise, the JSON object is written to the response body and the given custom handler is
+     * invoked (if not {@code null}).</li>
+     * </ol>
+     *
+     * @param ctx The routing context of the request.
+     * @param result The generic result of the operation.
+     * @param customHandler An (optional) handler for post processing the HTTP response, e.g. to set any additional HTTP
+     *                      headers. The handler <em>must not</em> write to response body. May be {@code null}.
+     * @param span The active OpenTracing span for this operation. The status of the response is logged and span is finished.
+     */
+    public static final void writeResponse(final RoutingContext ctx, final Result<?> result, final Handler<HttpServerResponse> customHandler, final Span span) {
+        final int status = result.getStatus();
+        final HttpServerResponse response = ctx.response();
+        response.setStatusCode(status);
+        if (status >= 400) {
+            HttpUtils.setResponseBody(response, JsonObject.mapFrom(result.getPayload()));
+        } else if (status == HttpURLConnection.HTTP_CREATED) {
+            HttpUtils.setResponseBody(response, JsonObject.mapFrom(result.getPayload()));
+            if (customHandler != null) {
+                customHandler.handle(response);
+            }
+        }
+        span.log(String.format("request finished with status: %s", status));
```

Comment:
I actually had in mind to set the `Tags.HTTP_STATUS` tag on the span. But maybe that is already done somewhere else? Who is actually creating the Span that is passed in here? Is this the span that is created by the `TracingHandler` that is registered by the `DeviceRegistryRestServer`?

Reply:
Yeah, that's the one and I don't see it being set anywhere else in this case. Fixed now